torch/_inductor/decomposition.py

```diff
@@ -680,3 +680,34 @@ def put(self, index, source, accumulate=False):
 def put_(self, index, source, accumulate=False):
     out = aten.put(self, index, source, accumulate=accumulate)
     return self.copy_(out)
+
+
+@register_decomposition(aten.index_reduce)
+def index_reduce(
+    self, dim: int, index, src, reduction_type: str, *, include_self: bool = True
+):
+    repeats = self.shape[dim + 1 :].numel() * self.shape[:dim].numel()
+    index_shape = (index.numel(), *self.shape[dim + 1 :], *self.shape[:dim])
+    perm = (*range(self.ndim - dim, self.ndim), 0, *range(1, self.ndim - dim))
+    scatter_index = index.repeat_interleave(repeats).reshape(index_shape).permute(perm)
+    return self.scatter_reduce(
+        dim,
+        scatter_index.to(torch.int64),
+        src,
+        reduction_type,
+        include_self=include_self,
+    )
+
+
+@register_decomposition(aten.index_reduce_)
+def index_reduce_(
+    self, dim: int, index, src, reduction_type: str, *, include_self: bool = True
+):
+    out = self.index_reduce(
+        dim,
+        index,
+        src,
+        reduction_type,
+        include_self=include_self,
+    )
+    return self.copy_(out)
```

Comment:
I think this should already be covered by this existing decomp:
https://github.com/pytorch/pytorch/blob/c7bf5871ceb02321ec8d8fd5fd305ad9c2fd8132/torch/_decomp/decompositions.py#L4783

Reply:
Right, thanks for pointing it out!